src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanParse.cs

```diff
@@ -563,20 +566,22 @@ internal bool SetBadFormatSpecifierFailure(char? formatSpecifierCharacter = null
             }
         }
 
-        internal static long Pow10(int pow)
+        internal static int Pow10(int pow)
         {
-            return pow switch
-            {
-                0 => 1,
-                1 => 10,
-                2 => 100,
-                3 => 1000,
-                4 => 10000,
-                5 => 100000,
-                6 => 1000000,
-                7 => 10000000,
-                _ => (long)Math.Pow(10, pow),
-            };
+            Debug.Assert(pow >= 0);
+            Debug.Assert(pow <= MaxFractionDigits);
+            ReadOnlySpan<int> powersOfTen = [
+                1,
+                10,
+                100,
+                1000,
+                10000,
+                100000,
+                1000000,
+                10000000,
+            ];
+            Debug.Assert(powersOfTen.Length == MaxFractionDigits + 1);
+            return Unsafe.Add(ref MemoryMarshal.GetReference(powersOfTen), pow);
```

Comment:
We try to avoid non-necessary "skip bound checks" unsafe patterns. Why is this code better than the jump table?

Reply:
Ok, I saw it in a few other places that's why I went with it here. Changed it to regular use accessor instead.

Reply:
Ah sorry now I understand how jump table works. Will revert this whole Pow10 change.

Reply:
I think it's fine to use your version as well as it produces smaller code in general (but with a memory load while jump table doesn't have it), it's just that we should use a normal indexer access rather than the unsafe option

Reply:
It's an indexer access now, thanks for the feedback!